receiver/filereceiver/file_reader.go

```diff
@@ -6,48 +6,94 @@ package filereceiver // import "github.com/open-telemetry/opentelemetry-collecto
 import (
 	"bufio"
 	"context"
+	"encoding/binary"
 	"errors"
 	"fmt"
 	"io"
 	"os"
 
-	"go.opentelemetry.io/collector/consumer"
+	"github.com/klauspost/compress/zstd"
 	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
 // stringReader is the only function we use from *bufio.Reader. We define it
 // so that it can be swapped out for testing.
 type stringReader interface {
 	ReadString(delim byte) (string, error)
+	Read(buf []byte) (int, error)
+}
+
+type unmarshaler struct {
+	metricsUnm pmetric.Unmarshaler
+	tracesUnm  ptrace.Unmarshaler
+	logsUnm    plog.Unmarshaler
 }
 
 // fileReader
 type fileReader struct {
 	stringReader stringReader
-	unm          pmetric.Unmarshaler
-	consumer     consumer.Metrics
+	unmarshaler  unmarshaler
+	consumer     consumerType
 	timer        *replayTimer
 }
 
-func newFileReader(consumer consumer.Metrics, file *os.File, timer *replayTimer) fileReader {
-	return fileReader{
-		consumer:     consumer,
-		stringReader: bufio.NewReader(file),
-		unm:          &pmetric.JSONUnmarshaler{},
-		timer:        timer,
+func newFileReader(consumer consumerType, file *os.File, timer *replayTimer, format string, compression string) fileReader {
+	fr := fileReader{
+		consumer: consumer,
+		timer:    timer,
+	}
+
+	if compression == compressionTypeZSTD {
+		cr, _ := zstd.NewReader(file)
+		fr.stringReader = bufio.NewReader(cr)
+	} else { // no compression
+		fr.stringReader = bufio.NewReader(file)
```

Comment:
Is there a reason to use a string reader?

Reply:
No particular reason other than that was already what was being used by the receiver, so I am just extending this functionality to also be able to read chunks of data (instead of just lines). Is there a major drawback/concern with using the string reader you have in mind?